torch/testing/_internal/common_methods_invocations.py

```diff
@@ -146,7 +146,7 @@ def formatter(arg):
                 return {k: formatter(v) for k, v in arg.items()}
             elif is_iterable_of_tensors(arg):
                 return "TensorList[" + ", ".join(map(formatter, arg)) + "]"
-            elif is_iter(arg):  # Handle list, tuple or any iterable type
+            elif is_iter(arg) and not isinstance(arg, str):  # Handle list, tuple or any iterable type
```

Comment:
`str` is also iterable. So we stay in this branch forever without this condition.

Reply:
What about make it more explicit to just handle lists and tuples?

Reply:
That would be fine. But if it is any other iterable holding tensor, we will fall to `repr(arg)` and end up printing the whole tensor.

Reply:
I understand; but we can deal with that later

Reply:
Cool!